core/templates/services/html-length.service.ts

```diff
@@ -18,12 +18,19 @@
 
 export const CALCULATION_TYPE_WORD = 'word';
 export const CALCULATION_TYPE_CHARACTER = 'character';
+export const STRING_TYPE_HTML = 'htmlString';
+export const STRING_TYPE_NORMAL = 'normalString';
+export const STRING_TYPE_NORMAL_AND_HTML = 'both';
 
 // eslint-disable-next-line max-len
 const CUSTOM_TAG_REGEX = /<oppia-noninteractive-(?:math|image|link|collapsible|video|skillreview|tabs)[^>]*>/g;
 
 type CalculationType =
   typeof CALCULATION_TYPE_WORD | typeof CALCULATION_TYPE_CHARACTER;
+type StringType =
```

Comment:
@HardikGoyal2003 Don't introduce this new parameter. Treat everything that comes in as an HTML string, but have the definition of HTML string include regular strings (which would be just treated as textContent in the root node). Otherwise you still have to do validation here, e.g. if STRING_TYPE_HTML comes in and there are no HTML tags you will need to raise an error.

My suggestion here is simpler: for an HTML string that comes in, simply include the character/word count of anything outside the HTML tags as well. That's all.